File: Data Structures & Algorithms/three-integer-sum/submission-5.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        Arrays.sort(nums);

        for (int left = 0; left < nums.length - 2; left++) {
            // דלג על כפילויות של left
            if (left > 0 && nums[left] == nums[left - 1]) continue;

            int mid = left + 1;
            int right = nums.length - 1;

            while (mid < right) {
                int sum = nums[left] + nums[mid] + nums[right];

                if (sum == 0) {
                    result.add(Arrays.asList(nums[left], nums[mid], nums[right]));
                    // דלג על כפילויות של mid ו-right
                    while (mid < right && nums[mid] == nums[mid + 1]) mid++;
                    while (mid < right && nums[right] == nums[right - 1]) right--;
                    mid++;
                    right--;
                }
                else if (sum < 0) {
                    mid++;  // צריך להגדיל את הסכום
                }
                else {
                    right--;  // צריך להקטין את הסכום
                }
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/binary-search/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        left=0
        right=len(nums)-1

        while left <=right:
            midd=(left+right)//2

            if nums[midd]==target:
                return midd

            elif nums[midd]>target:
                right=midd-1
            
            elif nums[midd]<target:
                left=midd+1
        return -1

